extensions/sonarcloud/templates/netcore/template.json

```diff
@@ -0,0 +1,163 @@
+{
+  "id": "sonarcloud_netcore_template",
+  "name": ".NET Core with SonarCloud",
+  "canDelete": false,
+  "category": "Build",
+  "iconTaskId": "ce096e50-6155-4de8-8800-4221aaeed4a1",
+  "defaultHostedQueue": null,
+  "description": "Build .NET Core and ASP.Net Core applications, and analyze with SonarCloud.",
+  "template": {
+    "build": [
+      {
+        "environment": {},
+        "enabled": true,
+        "continueOnError": false,
+        "alwaysRun": false,
+        "displayName": "Restore",
+        "timeoutInMinutes": 0,
+        "task": {
+          "id": "5541a522-603c-47ad-91fc-a4b1d163081b",
+          "versionSpec": "2.*",
+          "definitionType": "task"
+        },
+        "inputs": {
+          "command": "restore",
+          "projects": "$(Parameters.RestoreBuildProjects)"
+        }
+      },
+      {
+        "environment": {},
+        "enabled": true,
+        "continueOnError": false,
+        "alwaysRun": false,
+        "displayName": "Prepare analysis on SonarCloud",
+        "timeoutInMinutes": 0,
+        "task": {
+          "id": "14d9cde6-c1da-4d55-aa01-2965cd301255",
+          "versionSpec": "1.*",
+          "definitionType": "task"
+        },
+        "inputs": {
+          "scannerMode": "MSBuild"
+        }
+      },
+      {
+        "environment": {},
+        "enabled": true,
+        "continueOnError": false,
+        "alwaysRun": false,
+        "displayName": "Build",
+        "timeoutInMinutes": 0,
+        "task": {
+          "id": "5541a522-603c-47ad-91fc-a4b1d163081b",
+          "versionSpec": "2.*",
+          "definitionType": "task"
+        },
+        "inputs": {
+          "command": "build",
+          "projects": "$(Parameters.RestoreBuildProjects)",
+          "arguments": "--configuration $(BuildConfiguration)"
+        }
+      },
+      {
+        "environment": {},
+        "enabled": true,
+        "continueOnError": false,
+        "alwaysRun": false,
+        "displayName": "Test",
+        "timeoutInMinutes": 0,
+        "task": {
+          "id": "5541a522-603c-47ad-91fc-a4b1d163081b",
+          "versionSpec": "2.*",
+          "definitionType": "task"
+        },
+        "inputs": {
+          "command": "test",
+          "projects": "$(Parameters.TestProjects)",
+          "arguments": "--configuration $(BuildConfiguration)"
+        }
+      },
+      {
+        "environment": {},
+        "enabled": true,
+        "continueOnError": false,
+        "alwaysRun": false,
+        "displayName": "Run Code Analysis",
+        "timeoutInMinutes": 0,
+        "condition": "succeeded()",
+        "task": {
+          "id": "ce096e50-6155-4de8-8800-4221aaeed4a1",
+          "versionSpec": "1.*",
+          "definitionType": "task"
+        },
+        "inputs": {}
+      },
+      {
+        "environment": {},
+        "enabled": false,
+        "continueOnError": false,
+        "alwaysRun": false,
+        "displayName": "Publish Quality Gate Result",
```

Comment:
Same than for inputs, what if you don't specify a displayName?

Reply:
It seems to use the _friendlyName_ from the _task.json_. I've removed the _displayName_ for all of our tasks.
I've left the display names for the Microsoft tasks in cases where the corresponding default VSTS template has a display name (i.e. for the NuGet restore step and _dotnet_ tasks).

Reply:
I would also remove for Microsoft tasks except if we see a benefit to not show the default friendly name. But that's minor.

Reply:
All of our templates are closely based on one of the default existing VSTS templates. I deliberately didn't change the definitions of the tasks copied from the standard templates to make it easier to compare our templates against the default templates in the future.

As it happens, the display name is required in three out of the four Microsoft tasks (the same task is run multiple times but with different parameters).